File: api/src/repositories/postgres/models/Patient.model.ts
import { Column, Entity, JoinColumn, OneToMany, PrimaryGeneratedColumn } from 'typeorm'

import { Characteristic } from '@/repositories/postgres/models'

@Entity('pacientes')
export class Patient {
  @PrimaryGeneratedColumn()
  id!: number

  @Column({ name: 'nome' })
  name!: string

  @Column({ name: 'idade' })
  age!: number

  @Column()
  cpf!: string

  @Column()
  rg!: string

  @Column({ name: 'data_nasc', type: 'date' })
  birthday!: string

  @Column({ name: 'sexo' })
  gender!: string

  @Column({ name: 'signo' })
  zodiacSign!: string

  @Column({ name: 'mae' })
  motherName!: string

  @Column({ name: 'pai' })
  fatherName!: string

  @Column()
  email!: string

  @Column({ name: 'senha' })
  password!: string

  @Column()
  cep!: string

  @Column({ name: 'endereco' })
  address!: string

  @Column({ name: 'numero' })
  number!: number

  @Column({ name: 'bairro' })
  district!: string

  @Column({ name: 'cidade' })
  city!: string

  @Column({ name: 'estado' })
  state!: string

  @Column({ name: 'telefone_fixo' })
  phone!: string

  @Column({ name: 'celular' })
  cellphone!: string

  @Column({ name: 'altura' })
  heigth!: string

  @Column({ name: 'peso' })
  weigth!: number

  @Column({ name: 'tipo_sanguineo' })
  bloodType!: string

  @Column({ name: 'cor' })
  color!: string

  @OneToMany(() => Characteristic, characteristic => characteristic.patient)
  @JoinColumn({ name: 'caracteristicas' })
  characteristics!: Characteristic[]
}

File: app/lib/models/patient_model.dart
import 'package:json_annotation/json_annotation.dart';

import 'characteristic_model.dart';

part 'patient_model.g.dart';

@JsonSerializable(explicitToJson: true)
class Patient {
  int id;
  String name;
  int age;
  String cpf;
  String rg;
  String birthday;
  String gender;
  String zodiacSign;
  String motherName;
  String fatherName;
  String email;
  String password;
  String cep;
  String address;
  String district;
  String city;
  String state;
  String phone;
  String cellphone;
  String bloodType;
  String color;
  List<Characteristic>? characteristics;

  Patient(
      {required this.id,
      required this.name,
      required this.age,
      required this.cpf,
      required this.rg,
      required this.birthday,
      required this.gender,
      required this.zodiacSign,
      required this.motherName,
      required this.fatherName,
      required this.email,
      required this.password,
      required this.cep,
      required this.address,
      required this.district,
      required this.city,
      required this.state,
      required this.phone,
      required this.cellphone,
      required this.bloodType,
      required this.color,
      required this.characteristics});

  factory Patient.fromJson(Map<String, dynamic> json) =>
      _$PatientFromJson(json);

  Map<String, dynamic> toJson() => _$PatientToJson(this);
}
